youtube_dl/extractor/youtube.py

```diff
@@ -1825,7 +1825,7 @@ def extract_player_response(player_response, video_id):
         # Get video info
         video_info = {}
         embed_webpage = None
-        if re.search(r'player-age-gate-content">', video_webpage) is not None:
+        if self._html_search_meta('og:restrictions:age', video_webpage) == "18+":
```

Comment:
Since your fix uses `_html_search_meta`, a non-fatal warning is sometimes now issued for non-age gated files:

    WARNING: unable to extract og:restrictions:age; please report this issue on https://yt-dl.org/bug ...

Changing your fix to the following eliminates the warning, and still seems to fix the age-gate problem #25954

    if re.search('og:restrictions:age', video_webpage) is not None:

Reply:
I fixed it by giving _html_search_meta a default of None